server/controllers/manage/premises/placements/nonArrivalsController.ts

```diff
@@ -0,0 +1,73 @@
+import { type Request, RequestHandler, type Response } from 'express'
+import { Cas1NonArrival, NonArrivalReason } from '@approved-premises/api'
+import { PlacementService, PremisesService } from '../../../../services'
+import { catchValidationErrorOrPropogate, fetchErrorsAndUserInput } from '../../../../utils/validation'
+import managePaths from '../../../../paths/manage'
+import { ValidationError } from '../../../../utils/errors'
+
+export default class NonArrivalsController {
+  constructor(
+    private readonly premisesService: PremisesService,
+    private readonly placementService: PlacementService,
+  ) {}
+
+  new(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { token } = req.user
+      const { premisesId, placementId } = req.params
+      const { errors, errorSummary, userInput, errorTitle } = fetchErrorsAndUserInput(req)
+      const placement = await this.premisesService.getPlacement({ token, premisesId, placementId })
+      const nonArrivalReasons: Array<NonArrivalReason> = await this.placementService.getNonArrivalReasons(token)
+      return res.render('manage/premises/placements/non-arrival', {
+        nonArrivalReasons: nonArrivalReasons.filter(({ isActive }) => isActive),
```

Comment:
Is this necessary? I believe the API filters inactive reference data out, so while the type specifies `isActive` as boolean it should never be `false` -- makes me feel I should really check this!

Reply:
Oh, OK. I kinda assumed that if the API was filtering them out, it wouldn't bother returning the flag. I'll test this - I think all the test data has the flag set true.

Reply:
If that's the case, that seems to be a job for the service rather than the controller? I don't think we'll ever make use of inactive items, so let's remove them as early as possible.

Reply:
OK